### PR TITLE
[WIP] Extend WelcomeActivityTest

### LINKED ISSUE #2644: Extend WelcomeActivityTest
**Summary:** 

Would be good to extend WelcomeActivityTest to cover more cases. These are taken from #2636:

- Test swiping the correct number of times and then clicking 'yes' exits WelcomeActivity
- ~Test that on beta clicking the 'skip tutorial' button exits WelcomeActivity~
- ~Test we can swipe both ways (i.e. backwards and forwards)~
- Test swiping beyond the proper bounds shouldn't do anything e.g. swiping left on the first page or right on the last page

### PR DESCRIPTION
**Pull request by vanshikaarora**
**Description (required)**

Fixes #2644 Extend WelcomeActivityTest

Test that on beta clicking the 'skip tutorial' button exits WelcomeActivity
Test we can swipe both ways (i.e. backwards and forwards)

Added the above three tests

**Tests performed (required)**

Tested betaDebug on vivo 1713 with API level 24.

### CODE REVIEW

File: app/src/androidTest/java/fr/free/nrw/commons/WelcomeActivityTest.kt [hunk 1]
```diff
@@ -34,4 +35,43 @@ class WelcomeActivityTest {
...
+                    .perform(ViewActions.swipeLeft())
+            assert(true)
+            onView(withId(R.id.welcomePager))
+                    .perform(ViewActions.swipeRight())
+            assert(true)
+        }
+    }
+
+    @Test
+    fun ifEndedTutorial() {
```
Comment (domdomegg) [inline app/src/androidTest/java/fr/free/nrw/commons/WelcomeActivityTest.kt]: Maybe rename this to something that better describes it - e.g. `testSwipingWholeTutorial`. Reason the methods already there we prefixed with 'if' was just that they are special cases (i.e. if beta, then it should show the skip button).

File: app/src/androidTest/java/fr/free/nrw/commons/WelcomeActivityTest.kt [hunk 2]
```diff
@@ -34,4 +35,43 @@ class WelcomeActivityTest {
...
+        if (!ConfigUtils.isBetaFlavour()) {
```
Comment (domdomegg) [inline app/src/androidTest/java/fr/free/nrw/commons/WelcomeActivityTest.kt]: I think this test should work whether we are on beta or not so we can remove this if condition.

File: app/src/androidTest/java/fr/free/nrw/commons/WelcomeActivityTest.kt [hunk 3]
```diff
@@ -34,4 +35,43 @@ class WelcomeActivityTest {
...
+    fun ifSkipButtonClicked() {
+        if (!ConfigUtils.isBetaFlavour()) {
+            onView(withId(R.id.finishTutorialButton))
+                    .perform(ViewActions.click())
+            assert(activityRule.activity.isDestroyed)
+        }
+    }
+
+    @Test
+    fun ifSwiped() {
```
Comment (domdomegg) [inline app/src/androidTest/java/fr/free/nrw/commons/WelcomeActivityTest.kt]: Maybe rename to `testSwipingOnce` / similar

File: app/src/androidTest/java/fr/free/nrw/commons/WelcomeActivityTest.kt [hunk 4]
```diff
@@ -34,4 +35,43 @@ class WelcomeActivityTest {
...
+        if (!ConfigUtils.isBetaFlavour()) {
```
Comment (domdomegg) [inline app/src/androidTest/java/fr/free/nrw/commons/WelcomeActivityTest.kt]: I think this test should work whether we are on beta or not so we can remove this if condition.

File: app/src/androidTest/java/fr/free/nrw/commons/WelcomeActivityTest.kt [hunk 5]
```diff
@@ -34,4 +35,43 @@ class WelcomeActivityTest {
                     .check(matches(not(isDisplayed())))
         }
     }
+
+    @Test
+    fun ifSkipButtonClicked() {
+        if (!ConfigUtils.isBetaFlavour()) {
```
Comment (domdomegg) [inline app/src/androidTest/java/fr/free/nrw/commons/WelcomeActivityTest.kt]: I think this if statement is the wrong way round - we only want to check the skip button if we are on beta (i.e. this line needs the `!` removed).

**Comment by vanshikaarora** [comment]
>as that breaks when running tests in the prod build variant

@domdomegg I could't reproduce this. Can you please share the error report :)

**Comment by domdomegg** [comment]
> @domdomegg I could't reproduce this. Can you please share the error report :)

You fixed this in your latest commit already :)

Will review and merge later hopefully.

**Pull request merged**